submissions/Roophee/a-tiny-js-world-oop/index.js

```diff
@@ -0,0 +1,85 @@
+"use strict";
+
+class Animalia {
+    alive = true;
+    canSaying = false;
+    constructor(gender, legs, tail) {
+        this.legs = legs;
+        this.tail = tail;
+        this.gender = gender;
+    }
+}
+
+class Mammalia extends Animalia {
+    canSaying = true;
+    gregarious = true;
+    description (arrayOfProperties) {
+        return arrayOfProperties.map(property =>this[property]).join(';')
+    }
+    constructor(gender, friends, legs, tail) {
+        super(gender, legs, tail);
+        this.friends = friends.split(',');
+
+    }
+}
+
+class HomoSapiens extends Mammalia {
+    species = "human";
+    hasName = true;
+    constructor(name, gender, friends, sayingText) {
+        super(gender, friends);
+        this.name = name;
+        this.legs = 2;
+        this.hands = 2;
+        this.tail = 'none';
+        this.saying = sayingText;
+    }
+}
+
+class Dog extends Mammalia {
+    species = "dog";
+    hasName = true;
+    static saying = "Woof-Woof!";
+    constructor(name, gender, friends) {
+        super(gender, friends);
```

Comment:
`super` here calls constructor of `Mammalia` that generously offers to send it also legs and tail.
Why not using it?
Same is applicable to many other props.